src/debugger/buildConfig.ts

```diff
@@ -387,37 +422,82 @@ export class TestingDebugConfigurationFactory {
         return isRelease(this.testKind) ? "release" : "debug";
     }
 
-    private get xcTestOutputPath(): string {
+    private xcTestOutputPath(): string {
         return path.join(
             this.buildDirectory,
             this.artifactFolderForTestKind,
-            this.ctx.swiftPackage.name + "PackageTests.xctest"
+            `${this.ctx.swiftPackage.name}PackageTests.xctest`
         );
     }
 
-    private get swiftTestingOutputPath(): string {
+    private swiftTestingOutputPath(): string {
         return path.join(
             this.buildDirectory,
             this.artifactFolderForTestKind,
             `${this.ctx.swiftPackage.name}PackageTests.swift-testing`
         );
     }
 
-    private get testExecutableOutputPath(): string {
+    private async isUnifiedTestingBinary(): Promise<boolean> {
+        // Toolchains that contain https://github.com/swiftlang/swift-package-manager/commit/844bd137070dcd18d0f46dd95885ef7907ea0697
+        // no longer produce a .swift-testing binary, instead we want to use `unifiedTestingOutputPath`.
+        // In order to determine if we're working with a unified binary we need to check if the .swift-testing
+        // binary _doesn't_ exist, and if it does exist we need to check that it wasn't built by an old toolchain
+        // and is still in the .build directory. We do this by checking its mtime and seeing if it is after
+        // the `buildStartTime`.
+
+        // TODO: When Swift 6 is released and enough time has passed that we're sure no one is building the .swift-testing
+        //       binary anymore this fs.stat workaround can be removed and `swiftTestingPath` can be returned. The
+        //       `buildStartTime` argument can be removed and the build config generation can be made synchronous again.
+
+        try {
+            const stat = await fs.stat(this.swiftTestingOutputPath());
+            return !this.buildStartTime || stat.mtime.getTime() < this.buildStartTime.getTime();
+        } catch {
+            // If the .swift-testing binary doesn't exist then swift-testing tests are in the unified binary.
```

Comment:
This may not be true if `swift test --disable-experimental-swift-testing` had previously been passed.

Reply:
It shouldn't be possible to get in to this situation since the last run command will be one that we control that either does or does not build the unified test binary. This code path only happens immediately after a `swift build --build-tests` has run in VS Code, and we're doing a test run with swift-testing tests.